Loader/SymfonyFixturesLoader.php

```diff
@@ -35,15 +50,18 @@ public function addFixture(FixtureInterface $fixture)
      */
     protected function createFixture($class)
     {
-        try {
-            /*
-             * We don't actually need to create the fixture. We just
-             * return the one that already exists.
-             */
-            return $this->getFixture($class);
-        } catch (\InvalidArgumentException $e) {
-            throw new \LogicException(sprintf('The "%s" fixture class is trying to be loaded, but is not available. Make sure this class is defined as a service and tagged with "%s".', $class, FixturesCompilerPass::FIXTURE_TAG));
+        /*
+         * We don't actually need to create the fixture. We just
+         * return the one that already exists.
+         */
+
+        if (!isset($this->loadedFixtures[$class])) {
```

Comment:
This forbids using a fixture added using `addFixture` as the dependency of another fixture as it is not added to `loadedFixtures` there. `addFixture` should add the given fixture to the `loadedFixtures` array unless it's already present. In that case, there should be an exception if the fixture objects are not the same as a sanity check.

Reply:
Fixed by adding 
```php
$class = get_class($fixture);
        if (!isset($this->loadedFixtures[$class])) {
            $this->loadedFixtures[$class] = $fixture;
        }
``` 
to `addFixture`

Would that solve it?